src/tasks/dump.ts

```diff
@@ -81,13 +83,75 @@ interface DumpInstructions {
   transferToReceiver?: TransferToReceiver;
 }
 
+interface Receiver {
+  address: string;
+  isSameAsUser: boolean;
+}
+
+interface GetTransferToReceiverInput {
+  toToken: Erc20Token | NativeToken;
+  inputDumpedTokens: string[];
+  user: string;
+  receiver: Receiver;
+  maxFeePercent: number;
+  network: SupportedNetwork;
+  api: Api;
+  hre: HardhatRuntimeEnvironment;
+}
+async function getTransferToReceiver({
+  toToken,
+  inputDumpedTokens,
+  user,
+  receiver,
+  maxFeePercent,
+  network,
+  api,
+  hre,
+}: GetTransferToReceiverInput): Promise<TransferToReceiver | undefined> {
+  if (
+    receiver.isSameAsUser ||
+    !inputDumpedTokens.includes(
+      isNativeToken(toToken) ? BUY_ETH_ADDRESS : toToken.address,
+    )
+  ) {
+    return undefined;
+  }
+
+  const gasPrice = hre.ethers.provider.getGasPrice();
+  const amount = await balanceOf(toToken, user);
+  if (amount.isZero()) {
+    gasPrice.catch(() => {
+      // Gas price is not needed. Empty catch to suppress Node warning
+      // (unhandled promise rejection) in case retrieving gas fails
+    });
```

Comment:
This is a result of trying to speed things up by fetching the gas price and balance in parallel right? Why not move this down and make:

```
  const [gasPrice, gas, value] = await Promise.all([
    hre.ethers.provider.getGasPrice(),
    estimateTransferGas(toToken, user, receiver.address, amount),
    ethValue(toToken, amount, network, api),
  ]);
```

Reply:
When I wrote this I noticed that the two requests inside `Promise.all` need the balance, however retrieving the gas price doesn't need anything and can be executed while first the balance and then the other two requests are fetched. On the other hand, it's very unlikely that this optimization has any actual impact on the time taken by the script, so I went for your suggestion.